mne/coreg.py

```diff
@@ -1223,3 +1227,576 @@ def _scale_xfm(subject_to, xfm_fname, mri_name, subject_from, scale,
+            else:
+                self._update_params(est)
+            angle, move, scale = self._changes
+            if angle <= self._icp_angle and move <= self._icp_distance and \
+                    all(scale <= self._icp_scale):
+                break
+
+    def omit_hsp_points(self, distance):
+        """Exclude head shape points that are far away from the MRI head.
+
+        Parameters
+        ----------
+        distance : float
+            Exclude all points that are further away from the MRI head than
+            this distance. A value of distance <= 0 excludes nothing.
+        """
+        distance = float(distance)
+        if distance <= 0:
+            return
+
+        # find the new filter
+        mask = self._orig_hsp_point_distance <= distance
+        n_excluded = np.sum(~mask)
+        logger.info("Coregistration: Excluding %i head shape points with "
+                    "distance >= %.3f m.", n_excluded, distance)
+        # set the filter
+        with warnings.catch_warnings(record=True):  # comp to None in Traits
+            self._dig.extra_points_filter = mask
+
+    def point_distance(self):
+        """Compute Euclidean distance between the head-mri points."""
+        mri_points = list()
+        hsp_points = list()
+        if self._hsp_weight > 0 and self._has_dig_data:
+            mri_points.append(self._transformed_high_res_mri_points[
+                self._nearest_transformed_high_res_mri_idx_hsp])
+            hsp_points.append(self._transformed_dig_points)
+            assert len(mri_points[-1]) == len(hsp_points[-1])
+        if self._eeg_weight > 0 and self._has_eeg_data:
+            mri_points.append(self._transformed_high_res_mri_points[
+                self._nearest_transformed_high_res_mri_idx_eeg])
+            hsp_points.append(self._transformed_dig_eeg_points)
+            assert len(mri_points[-1]) == len(hsp_points[-1])
+        if self._hpi_weight > 0 and self._has_hpi_data:
+            mri_points.append(self._transformed_high_res_mri_points[
+                self._nearest_transformed_high_res_mri_idx_hpi])
+            hsp_points.append(self._transformed_dig_hpi)
+            assert len(mri_points[-1]) == len(hsp_points[-1])
+        if all(len(h) == 0 for h in hsp_points):
+            return None
+        mri_points = np.concatenate(mri_points)
+        hsp_points = np.concatenate(hsp_points)
+        return np.linalg.norm(mri_points - hsp_points, axis=-1)
+
+    @property
+    def trans(self):
+        """Return the head-mri transform."""
+        return Transform('head', 'mri', self._head_mri_t)
+
+    def reset(self):
+        """Reset all the parameters affecting the coregistration."""
+        self._n_scale_param = self._n_scale_params[0]
+        self._grow_hair = 0.
+        self._parameters[:] = list(self._default_parameters)
+        self._last_parameters[:] = list(self._default_parameters)
+        self.omit_hsp_points(np.inf)
+
+
+class _DigSource(object):
```

Comment:
do we really need the next 3 classes?

Reply:
Agreed we should get rid of this class and just use things like `mne.io.digitization._get_data_as_dict_from_dig`, I think it'll get us most of the way there.